Objects/floatobject.c

```diff
@@ -1683,8 +1684,97 @@ typedef enum _py_float_format_type float_format_type;
 #define ieee_big_endian_format _py_float_format_ieee_big_endian
 #define ieee_little_endian_format _py_float_format_ieee_little_endian
 
-#define float_format (_PyRuntime.float_state.float_format)
-#define double_format (_PyRuntime.float_state.double_format)
+#ifdef __GNUC__
+__attribute__((always_inline))
+#endif
```

Comment:
Don't use GNUC extensions. Leave it to the compiler instead.

Reply:
gcc seems to have problems with link time optimizations and this kind of inline.